allennlp/data/dataset_readers/wikitables/world.py

```diff
@@ -0,0 +1,143 @@
+    Sempre variables in all logical forms to NLTK variables, and the types of all predicates and entities.
+    """
+    def __init__(self, table_graph: TableKnowledgeGraph) -> None:
+        self.table_graph = table_graph
+        # NLTK has a naming convention for variable types (see ``_map_name`` for more details).
+        # We initialize this dict with common predicate names and update it as we process logical forms.
+        # TODO (pradeep): Should we do updates while reading tables instead?
+        self.name_mapping = COMMON_NAME_MAPPING.copy()
+        # For every new Sempre column name seen, we update this counter to map it to a new NLTK name.
+        self._column_counter = 0
+        # Initial type signature. Will update when we see more predicates.
+        self.type_signature = COMMON_TYPE_SIGNATURE.copy()
+        self._logic_parser = DynamicTypeLogicParser(type_check=True)
+
+    def process_sempre_forms(self, sempre_forms: List[str]) -> List[Expression]:
+        """
+        Processes logical forms in Sempre format (that are either gold annotations or output from DPD)
+        and converts them into ``Expression`` objects in NLTK. The conversion involves mapping names to
+        follow NLTK's variable naming conventions.
+        """
+        expressions = []
+        for sempre_form in sempre_forms:
+            parsed_lisp = pyparsing.OneOrMore(pyparsing.nestedExpr()).parseString(sempre_form).asList()
+            translated_string = self._process_nested_expression(parsed_lisp)
+            expression = self._logic_parser.parse(translated_string, signature=self.type_signature)
+            expressions.append(expression)
+        return expressions
+
+    def _process_nested_expression(self, nested_expression):
+        """
+        ``nested_expression`` is the result of parsing a Lambda-DCS expression in Lisp format. We process it
+        recursively and return a string in the format that NLTK's ``LogicParser`` would understand.
+        """
+        expression_is_list = isinstance(nested_expression, list)
+        expression_size = len(nested_expression)
+        if expression_is_list and expression_size == 1 and isinstance(nested_expression[0], list):
+            return self._process_nested_expression(nested_expression[0])
+        elements_are_leaves = [isinstance(element, str) for element in nested_expression]
+        if all(elements_are_leaves):
+            mapped_names = [self._map_name(name) for name in nested_expression]
+        else:
+            mapped_names = []
+            for element, is_leaf in zip(nested_expression, elements_are_leaves):
+                if is_leaf:
+                    mapped_names.append(self._map_name(element))
+                else:
+                    mapped_names.append(self._process_nested_expression(element))
+        if mapped_names[0] == "\\":
+            # This means the predicate is lambda. NLTK wants the variable name to not be within parantheses.
```

Comment:
Aren't you adding parentheses here?  Is there an extra "not" in this comment?

Reply:
I'm adding parentheses only after the variable (as opposed to what's happening in else). I'll make the comment clearer.